cps/src/main/java/org/jenkinsci/plugins/workflow/cps/steps/LoadStepExecution.java

```diff
@@ -27,14 +34,22 @@
     @StepContextParameter
     private transient FlowNode node;
 
+    @StepContextParameter
+    private transient TaskListener listener;
+
     @Override
     public boolean start() throws Exception {
         CpsStepContext cps = (CpsStepContext) getContext();
         CpsThread t = CpsThread.current();
 
-        GroovyShell shell = t.getExecution().getShell();
+        CpsFlowExecution execution = t.getExecution();
 
-        Script script = shell.parse(cwd.child(step.getPath()).readToString());
+        String text = cwd.child(step.getPath()).readToString();
+        for (Replacer replacer : ExtensionList.lookup(Replacer.class)) {
```

Comment:
Is there a reason why it wouldn't make more sense for the Replacer to do a bulk replacement of scripts?  We could still chain them if needed, I suppose.